Sheet by month

horknfbr opened the pull request:
I had made some local changes to the script and figured I would go a bit further and submit it as a pull request.

Mainly this creates a new sheet for each month. I've also aded it as a cli switch, as well as cli args for sheetname and workbook name.

Commit:
Removed unused modules

refactored code quite a bit

added in ability to create a new sheet for each month

added in function for testing, will be removed

Commit:
finished logic for adding header row to new sheet if created
changed date format to be auto detected by google sheets as a date
refactored code for readability

Commit:
added new usage information

Commit:
Adding in paramaters for nameing the Worrkbook, naming the Sheeet, and creating a sheet per month

frdmn commented:
Hi @horknfbr - many thanks for your contribution.

I did a couple of changes the past days which now caused some conflicts in this PR. Would you mind rebasing from master and trying to get rid of them so I can merge this pull request? :)

Commit:
rebased with master, changed values to Mbps using powers of 2
broke up print line to stay under 72 char limit recommended in flake8

Commit:
moving results print outside of Function

Commit:
fixed issues with the rebase

Commit:
fixed a random capitalization

horknfbr commented:
should be all set, i changed the math on the conversion to Mbits to use powers of two, let me know if this is ok, i can revert to 1000/1000 if preferable.

frdmn commented:
Can you check if `-b` is _not_ default when not specified?

I just pulled the changes from this PR - then just executed the script without any arguments:

```
$ python3 ./speedtest-charts.py
```

... but it still created a new sheet (as If I did specify the `-b` switch - I didn't):

![ohnetitel mov](https://user-images.githubusercontent.com/944459/42759507-80fdc1d6-8907-11e8-973f-0d1ae1ecbef0.gif)

Or am I missing something?

One last thing - since we're talking about bits and not bytes - we have to divide by 1000 rather than 1024. Could you change that?

Commit:
fixed bad logic on witing to sheet is -b not selected

chnaged math on mbit to be bas 1000 instead on base 1024

Commit:
removed sheetname cli arg as it is not really needed

refactored the logic to only write to first sheet if -b not set

horknfbr commented:
@frdmn sorry about that, I made a bad assumption that the first sheet would be named "Sheet 1"  the code now uses the correct function to append to the first sheet in the workbook if -b is not set on the command line.

I also switched the math back to base 1000

I removed the cli option to name the sheet since we default to writing to the first sheet, so name is not really needed

frdmn commented:
Looks superb! Thanks Scott :)
